src/main/java/bio/terra/common/retry/transaction/TransactionRetryInterceptorFactory.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.common.retry.transaction;
+
+import bio.terra.common.retry.CompositeBackOffPolicy;
+import java.util.LinkedHashMap;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.classify.BinaryExceptionClassifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.backoff.UniformRandomBackOffPolicy;
+import org.springframework.retry.interceptor.RetryInterceptorBuilder;
+import org.springframework.retry.policy.CompositeRetryPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+
+@Configuration
+@EnableConfigurationProperties(TransactionRetryConfig.class)
+public class TransactionRetryInterceptorFactory {
+  /**
+   * Creates an interceptor that can be used in {@link
+   * org.springframework.retry.annotation.Retryable}: <code>
+   * @Retryable(interceptor = "transactionRetryInterceptor")</code>. Be sure to use {@link
+   * org.springframework.retry.annotation.EnableRetry}.
+   */
```

Comment:
What is the cardinality of this? Factory makes it sound like there is more than one. So is it done once for each `@ReadTransaction` or `@WriteTransaction` or is it done on each entry into the retry code?

Reply:
I think it is just once. What do you call a class that has a method with a @Bean annotation?